proto/spacemesh/v1/app_types.proto

```diff
@@ -0,0 +1,129 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+import "spacemesh/v1/types.proto";
+
+// Spacemesh app primitive variable types
+enum VariableType {
+    VARIABLE_TYPE_UNSPECIFIED = 0;
+    VARIABLE_TYPE_I8 = 1;
+    VARIABLE_TYPE_U8 = 2;
+    VARIABLE_TYPE_I16 = 3;
+    VARIABLE_TYPE_U16 = 4;
+    VARIABLE_TYPE_I32 = 5;
+    VARIABLE_TYPE_U32 = 6;
+    VARIABLE_TYPE_I64 = 7;
+    VARIABLE_TYPE_U64 = 8;
+    VARIABLE_TYPE_BOOL = 9;
+    VARIABLE_TYPE_AMOUNT = 10;
+    VARIABLE_TYPE_ADDRESS = 11;
+    // future supported types will be added here
+}
+
+// Variable metadata
+message VariableMetadata {
+    uint32 id = 1; // var index in app's state
+    string name = 2; // var name
+    VariableType type = 3; // type
+    bool is_array = 4; // when true, variable is an array of 0 or more primitive value types, otherwise, a single primitive type value.
+}
+
+message ValuesArray {
+    repeated bytes values = 1;
+}
+
+// The meta-data and value of an app's variable
+message Variable {
+    VariableMetadata meta_data = 1;
+    oneof data {
+        bytes value = 2; // since bytes size is arbitrary - this will support future var types
+        ValuesArray values = 3;
+    }
+}
+
+// The address of an app template in the templates global state store
+message TemplateAddress {
+    bytes address = 1;
+}
+
+// Information about an app template in the global state store
+message Template {
+    TemplateAddress address = 1; // template address
+    string name = 2; // template name
+
+    // note - the author can be obtained from the deploy transaction but it is nice to have it here
+    AccountId author = 3; // template uploader
```

Comment:
```suggestion
    AccountId deployer = 3; // template uploader
```

Reply:
I'm using the svm terminology as specified by yaron, until we discuss making any changes to this terminology with yaron.